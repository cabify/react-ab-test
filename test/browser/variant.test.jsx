import React from 'react';
import { mount } from 'enzyme';

import CoreExperiment from '../../src/CoreExperiment.jsx';
import Variant from '../../src/Variant.jsx';
import emitter from '../../src/emitter.jsx';

describe('Variant', () => {
  it('should render text nodes', () => {
    const variantTextA = 'variantTextA';
    const variantTextB = 'variantTextB';

    const wrapper = mount(
      <CoreExperiment name="text-nodes" defaultVariantName="A">
        <Variant name="A">{variantTextA}</Variant>
        <Variant name="B">{variantTextB}</Variant>
      </CoreExperiment>
    );

    expect(
      wrapper
        .find(Variant)
        .at(0)
        .text()
    ).toBe(variantTextA);
  });

  it('should render components', () => {
    const wrapper = mount(
      <CoreExperiment name="components" defaultVariantName="A">
        <Variant name="A">
          <div id="variant-a" />
        </Variant>
        <Variant name="B">
          <div id="variant-b" />
        </Variant>
      </CoreExperiment>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render arrays of components', () => {
    const wrapper = mount(
      <CoreExperiment name="array-of-elements" defaultVariantName="A">
        <Variant name="A">
          <div id="variant-a" />
          <div />
        </Variant>
        <Variant name="B">
          <div id="variant-b" />
          <div />
        </Variant>
      </CoreExperiment>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
