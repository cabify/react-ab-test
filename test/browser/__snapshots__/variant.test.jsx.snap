// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variant should render arrays of components 1`] = `
<CoreExperiment
  defaultVariantName="A"
  name="array-of-elements"
>
  <Pushtell.Variant
    name="A"
  >
    <div>
      <div
        id="variant-a"
      />
      <div />
    </div>
  </Pushtell.Variant>
</CoreExperiment>
`;

exports[`Variant should render components 1`] = `
<CoreExperiment
  defaultVariantName="A"
  name="components"
>
  <Pushtell.Variant
    name="A"
  >
    <div
      id="variant-a"
    />
  </Pushtell.Variant>
</CoreExperiment>
`;
